carla/evaluation/benchmark.py

```diff
@@ -101,10 +102,27 @@ def compute_distances(self) -> pd.DataFrame:
         factual_without_nans, counterfactuals_without_nans = remove_nans(
             self._enc_norm_factuals, self._counterfactuals
         )
-        arr_f = factual_without_nans.to_numpy()
-        arr_cf = counterfactuals_without_nans.to_numpy()
 
-        distances = get_distances(arr_f, arr_cf)
+        if counterfactuals_without_nans.empty:
+            distances = np.nan
+        else:
+            if self._mlmodel.encoder.drop is None:
+                # To prevent double count of encoded features without drop if_binary
+                binary_columns_to_drop = get_drop_columns_binary(
+                    self._mlmodel.data.categoricals,
+                    counterfactuals_without_nans.columns.tolist(),
+                )
+                counterfactuals_without_nans = counterfactuals_without_nans.drop(
+                    binary_columns_to_drop, axis=1
+                )
+                factual_without_nans = factual_without_nans.drop(
+                    binary_columns_to_drop, axis=1
+                )
```

Comment:
If we return early here we can avoid the nested else condition
```suggestion
        if counterfactuals_without_nans.empty:
            return pd.DataFrame(np.nan, columns=columns)
            
        if self._mlmodel.encoder.drop is None:
            # To prevent double count of encoded features without drop if_binary
            binary_columns_to_drop = get_drop_columns_binary(
                self._mlmodel.data.categoricals,
                counterfactuals_without_nans.columns.tolist(),
            )
            counterfactuals_without_nans = counterfactuals_without_nans.drop(
                binary_columns_to_drop, axis=1
            )
            factual_without_nans = factual_without_nans.drop(
                binary_columns_to_drop, axis=1
            )
```

Reply:
Ah you said that multiple now, sorry. Will pay attention to that in future.